src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -32,56 +32,63 @@
 
   The keys should have the following values:
 
-  * :ruby-load-path - a vector of file paths, containing the locations of puppet source code.
-
-  * :gem-home - The location that JRuby gems are stored
-
-  * :master-conf-dir - file path to puppetmaster's conf dir;
-    if not specified, will use the puppet default.
-
-  * :master-code-dir - file path to puppetmaster's code dir;
-    if not specified, will use the puppet default.
-
-  * :max-active-instances - The maximum number of JRubyPuppet instances that
-      will be pooled.
-  
-  * :master-var-dir - path to the puppetmaster's var dir;
-    if not specified, will use the puppet default.
-
-  * :master-run-dir - path to the puppetmaster's run dir;
-    if not specified, will use the puppet default.
-
-  * :master-log-dir - path to the puppetmaster's log dir;
-    if not specified, will use the puppet default.
-
-  * :max-active-instances - The maximum number of JRubyPuppet instances that
-      will be pooled. If not specified, the system's
-      number of CPUs+2 will be used.
-
-  * :http-client-ssl-protocols - A list of legal SSL protocols that may be used
-      when https client requests are made.
-
-  * :http-client-cipher-suites - A list of legal SSL cipher suites that may
-      be used when https client requests are made."
-  {
-   ;; NOTE: there is a bug in the version of schema we're using, which causes
-   ;; the order of things that you put into a `both` to be very important.
-   ;; The `vector?` pred here MUST come before the `[schema/Str]`.  For more info
-   ;; see https://github.com/Prismatic/schema/issues/68
-   :ruby-load-path                           (schema/both (schema/pred vector?) [schema/Str])
-   :gem-home                                 schema/Str
-   :master-conf-dir                          (schema/maybe schema/Str)
-   :master-code-dir                          (schema/maybe schema/Str)
-   :master-var-dir                           (schema/maybe schema/Str)
-   :master-run-dir                           (schema/maybe schema/Str)
-   :master-log-dir                           (schema/maybe schema/Str)
-   :http-client-ssl-protocols                [schema/Str]
-   :http-client-cipher-suites                [schema/Str]
+    * :ruby-load-path - a vector of file paths, containing the locations of puppet source code.
```

Comment:
Sorry for all of the spacing differences here.  Alignment and order was slightly different between the stable and master branches for these so the merge conflicts were hard to fix up without doing this.  On the plus side, hopefully any conflicts around these in the future should be easier to reconcile.

Highlights on the real differences, ignoring the spacing:

1) `:max-active-instances` was moved below `:master-log-dir` and the line within `:max-active-instances` *"If not specified, the system's number of CPUs+2 will be used"* was removed.

2) `:http-client-connect-timeout-milliseconds` and `:http-client-idle-timeout-milliseconds` text was added - should reflect stable branch implementation verbatim.